src/main/java/edu/ucla/library/bucketeer/Item.java

```diff
@@ -164,9 +165,7 @@ public Item setAccessURL(final String aAccessURL) {
         final String filePath;
         final File file;
 
-        if (myFilePath == null) {
-            return Optional.empty();
-        } else if (!hasFile() || myFilePath.isEmpty()) {
+        if (myFilePath == null || !hasFile() || myFilePath.isEmpty()) {
```

Comment:
Since the constructors both set `myFilePath` now, the only way this could be null is if someone passes a null to `setFilePath(final Optional<String> aFilePath)`. I'd be inclined to check for null in that setter rather than having to check for null here (that localizes the exception at the point of the problem (i.e. passing a null)). In the setter you could use `Objects.requireNonNull(aFilePath)` or the version that takes a string message. Then, here, you only have to check `hasFile()` and `myFilePath.isEmpty()`.